src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx

```diff
@@ -110,6 +122,55 @@ it("displays the artwork title correctly without date", () => {
   expect(serifChildren(component)).not.toContain(`${saleArtwork.artwork.title},`)
 })
 
+it("can load and display price summary", () => {
```

Comment:
Have we considered / would it make sense to consider what happens when the MP query for the `calculatedCost` field fails? I guess this could happen when:

- MP is down. Presumably a lot more would fail.
- There's a bug within the code called from querying for `calculatedCost` (seems more likely)

Would it be testing the case that we have some form of error from MP to ensure our front-end is resilient to reasonable backend errors? Thoughts?

Reply:
The `renderWithLoadProgress` also adds error handling and presenting a general error message. That said the `<LoadFailureView>` is designed to fit in a larger viewport and it gets squashed on a smaller screen like iPhone SE, and it is definitely something we should fix before release.

I've created a ticket for this https://artsyproduct.atlassian.net/browse/AUCT-674 on JIRA, but I would like us to be more iterative than making the implementation perfect in the first place.

Reply:
You linked to the main story, which you didn't create, but perhaps you meant to link to either:

- https://artsyproduct.atlassian.net/browse/AUCT-692 or
- https://artsyproduct.atlassian.net/browse/AUCT-695

How are these issues different btw? Is AUCT-692 strictly related to the visuals of the loading animation, while AUCT-699 is strictly related to the behavior when the endpoint is down?

---

Re being iterative: 9 times out of 10, I think we should totally optimize for iterative PRs and releases. I'm just being a bit cautious here as I know that there's no guarantee that iOS users update to the latest release.

So when you write:

> but I would like us to be more iterative than making the implementation perfect in the first place.

Are you suggesting that we allow a release an emission, but ensure we don't deploy Eigen until we resolved the mentioned issues? If so, totally down, but please communicate to Ash and the mobile experience team that we'll want the equivalent of a "deploy block" for Eigen until we resolve (to avoid the scenario of deploying emission mid-build). Perhaps this, in and of itself, is a good reason to have a lab feature here (as it would remove the need for such a deploy block).

Reply:
We want a feature flag like the following:

`if echo's feature flag is true OR the lab feature is enabled, then display the PriceSummary`.

this would allow us to:

1. deploy emission to eigen while we're working on this feature across multiple PRs as the lab feature is off by default, while simultaneously allowing us to.
2. Enable for _all users_ when we want to by turning on an echo flag.

---

- We are not going to handle the error states changes in this PR
- @yuki24 will create and work on a ticket to implement the echo based feature flag as a next up item.
  - ideally, we wouldn't deploy eigen until we have this echo based feature flag.

Reply:
> Perhaps this, in and of itself, is a good reason to have a lab feature here (as it would remove the need for such a deploy block).

If this ever becomes a release blocker, we could just remove it from the `<ComfirmBid>` component (literally a one-line change) and move it to the StoryBook. So I don't think Emission-only feature flagging still makes sense here. I don't think we should remove it at this point though since it's already there.